cirq/study/resolver.py

```diff
@@ -86,45 +87,41 @@ def value_of(self,
         Returns:
             The value of the parameter as resolved by this resolver.
         """
-        # Input is a float, no resolution needed: return early
-        if isinstance(value, float):
+        # Numerical inputs don't need resolution.
+        if isinstance(value, numbers.Number):
             return value
 
-        # Handles 2 cases:
-        # Input is a string and maps to a number in the dictionary
-        # Input is a symbol and maps to a number in the dictionary
-        # In both cases, return it directly.
-        if value in self.param_dict:
-            param_value = self.param_dict[value]
-            if isinstance(param_value, (float, int)):
-                return param_value
-
-        # Input is a string and is not in the dictionary.
-        # Treat it as a symbol instead.
+        # Strings are shorthands for symbols.
         if isinstance(value, str):
-            # If the string is in the param_dict as a value, return it.
-            # Otherwise, try using the symbol instead.
-            return self.value_of(sympy.Symbol(value))
-
-        # Input is a symbol (sympy.Symbol('a')) and its string maps to a number
-        # in the dictionary ({'a': 1.0}).  Return it.
-        if (isinstance(value, sympy.Symbol) and value.name in self.param_dict):
-            param_value = self.param_dict[value.name]
-            if isinstance(param_value, (float, int)):
-                return param_value
-
-        # Input is either a sympy formula or the dictionary maps to a
-        # formula.  Use sympy to resolve the value.
-        # Note that sympy.subs() is slow, so we want to avoid this and
-        # only use it for cases that require complicated resolution.
-        if isinstance(value, sympy.Basic):
+            value = sympy.Symbol(value)
+
+        # Handle cases where input is mapped to a resolved value.
+        key = None
+        if isinstance(value, sympy.Symbol):
+            if value in self.param_dict:
+                key = value
+            if value.name in self.param_dict:
+                key = value.name
+        if key is not None:
```

Comment:
This conditional should be indented one more level and `key = None` should be brought into `if isinstance(value, sympy.Symbol)`.